### PR TITLE
HL-120

### PR DESCRIPTION
**Pull request by JasminFarhi**
[[Link to ticket](url)](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-120)

- One liner ticket title/description
- If not intuitive from the ticket -> short summary of code changes
- Include screenshots for visual UI changes
Screen Shot 2022-11-22 at 5 37 30 pm

### CODE REVIEW

File: src/pages/Bets/Bets_Logic.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const BetsLogics = () => {
   const [myBetsEnabled, setMyBetsEnabled] = useState(true);
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [selectedBet, setSelectedBet] = useState<BetHistory>();
-
+  const [selectedFilter, setSelectedFilter] = useState<string>("ALL_BETS");
```
Comment (brettvniekerk) [inline src/pages/Bets/Bets_Logic.tsx:18]: import `FilterOption` and use that as the type for this - `string` is v open to dev error for this

File: src/pages/Bets/components/BetRows.tsx [hunk 2]
```diff
@@ -3,26 +3,65 @@ import { useAccount } from "wagmi";
 import { BetHistory } from "../../../types";
 import { Row } from "./BetTable";
 
+export type FilterOption = "ALL_BETS" | "RESULTED" | "PENDING" | "SETTLED";
+
+enum FilterOptions {
```
Comment (brettvniekerk) [inline src/pages/Bets/components/BetRows.tsx:8]: We should use the enum and not the type - not sure why there is two

File: src/pages/Bets/Bets_View.tsx [hunk 3]
```diff
@@ -13,6 +14,8 @@ type Props = {
   isModalOpen: boolean;
   onCloseModal: () => void;
   selectedBet?: BetHistory;
+  selectedFilter: string;
+  setSelectedFilter: (filter: string) => void;
```
Comment (brettvniekerk) [inline src/pages/Bets/Bets_View.tsx:18]: `filter: string` should be `filter: FilterOption`

File: src/pages/Bets/Bets_View.tsx [hunk 4]
```diff
@@ -44,53 +49,111 @@ const BetsView = ({
...
+        <h3 className="text-lg font-medium text-gray-900 flex items-center">
+          Bets History
+        </h3>
+
+        <div className="flex flex-row space-x-3">
+          <Button
+            className={classnames(
+              "cursor-pointer bg-indigo-600 sm:w-auto sm:mb-0",
+              {
+                "bg-indigo-800 sm:w-auto sm:mb-0": selectedFilter == "ALL_BETS"
```
Comment (brettvniekerk) [inline src/pages/Bets/Bets_View.tsx:63]: `selectedFilter == FilterOptions.ALL_BETS` pls

File: src/pages/Bets/Bets_View.tsx [hunk 5]
```diff
@@ -44,53 +49,111 @@ const BetsView = ({
...
+            )}
+            onClick={() => setSelectedFilter("ALL_BETS")}
+          >
+            All Bets
+          </Button>
+          <Button
+            className={classnames(
+              "cursor-pointer bg-indigo-600 sm:w-auto sm:mb-0",
+              {
+                "bg-indigo-800 sm:w-auto sm:mb-0": selectedFilter == "RESULTED"
```
Comment (brettvniekerk) [inline src/pages/Bets/Bets_View.tsx:77]: same enum here

File: src/pages/Bets/Bets_View.tsx [hunk 6]
```diff
@@ -44,53 +49,111 @@ const BetsView = ({
...
+            )}
+            onClick={() => setSelectedFilter("RESULTED")}
+          >
+            Resulted
+          </Button>
+          <Button
+            className={classnames(
+              "cursor-pointer bg-indigo-600  sm:w-auto sm:mb-0",
+              {
+                "bg-indigo-800 sm:w-auto sm:mb-0": selectedFilter === "PENDING"
```
Comment (brettvniekerk) [inline src/pages/Bets/Bets_View.tsx:91]: and here

File: src/pages/Bets/Bets_View.tsx [hunk 7]
```diff
@@ -44,53 +49,111 @@ const BetsView = ({
...
+            )}
+            onClick={() => setSelectedFilter("PENDING")}
+          >
+            Pending
+          </Button>
+          <Button
+            className={classnames(
+              "cursor-pointer bg-indigo-600 sm:w-auto sm:mb-0",
+              {
+                "bg-indigo-800 sm:w-auto sm:mb-0": selectedFilter === "SETTLED"
```
Comment (brettvniekerk) [inline src/pages/Bets/Bets_View.tsx:105]: and here

**Review by brettvniekerk — CHANGES_REQUESTED** [review]
Please refactor the filter types to the enum that's been provided - it will allow for stronger typing and reducing margin of dev error

**Review by OptimusOpus — DISMISSED** [review]

File: src/pages/Bets/components/BetRows.tsx [hunk 8]
```diff
@@ -3,26 +3,63 @@ import { useAccount } from "wagmi";
...
+            );
+          } else if (selectedFilter === FilterOptions.SETTLED && bet.settled) {
+            return (
+              <Row betData={bet} key={index} onClick={() => onClickBet(bet)} />
+            );
+          } else if (selectedFilter === FilterOptions.ALL_BETS) {
+            return (
+              <Row betData={bet} key={index} onClick={() => onClickBet(bet)} />
+            );
+          }
```
Comment (bitcoinbrisbane) [inline src/pages/Bets/components/BetRows.tsx:61]: I think the complexity can be reduced by not using `else if`

```suggestion
          } 
           if (
            selectedFilter === FilterOptions.PENDING &&
            !bet.winningPropositionId &&
            !bet.marketResultAdded &&
            !bet.settled
          ) {
            return (
              <Row betData={bet} key={index} onClick={() => onClickBet(bet)} />
            );
          } 
           if (selectedFilter === FilterOptions.SETTLED && bet.settled) {
            return (
              <Row betData={bet} key={index} onClick={() => onClickBet(bet)} />
            );
          } 
          if (selectedFilter === FilterOptions.ALL_BETS) {
            return (
              <Row betData={bet} key={index} onClick={() => onClickBet(bet)} />
            );
          }
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
See comments

**Review by OptimusOpus — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**